src/Data/Array/Polarized/Pull/Internal.hs

```diff
@@ -0,0 +1,73 @@
+{-# OPTIONS_GHC -fno-warn-partial-type-signatures #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+
+module Data.Array.Polarized.Pull.Internal where
+
+import qualified Data.Functor.Linear as Data
+import Prelude.Linear
+import qualified Prelude
+import Data.Vector (Vector)
+import qualified Data.Vector as Vector
+
+import qualified Unsafe.Linear as Unsafe
+
+-- | A pull array represents arrays which are easy to extract elements from.
+-- They provide an indexing function, and a length. If the indexing function
+-- is called at a point out of range, the result is unspecified.
+data Array a where
+  Array :: (Int -> a) -> Int -> Array a
+-- Int is movable, so this is the same as Array :: (Int -> a) -> ...
```

Comment:
These are syntactically the same thing. One of these you surely intended to be different. (probably the comment).

Reply:
Oops, yeah, it was meant to be a linear arrow in the comment